crates/fe/src/main.rs

```diff
@@ -296,3 +307,30 @@ fn mir_dump(input_path: &str) {
         std::process::exit(1)
     }
 }
+
+fn yul_functions_dump(input_path: &str) {
+    let mut db = fe_driver::NewDb::default();
+    if Path::new(input_path).is_file() {
+        let content = match std::fs::read_to_string(input_path) {
+            Err(err) => {
+                eprintln!("Failed to load file: `{}`. Error: {}", input_path, err);
+                std::process::exit(1)
+            }
+            Ok(content) => content,
+        };
+
+        match fe_driver::dump_codegen_funcs(&mut db, input_path, &content) {
+            Ok(contract) => {
+                println!("{}", contract)
+            }
+            Err(err) => {
+                eprintln!("Unable to dump mir `{}", input_path);
+                print_diagnostics(&db, &err.0);
+                std::process::exit(1)
+            }
+        }
+    } else {
+        eprintln!("mir doesn't support ingot yet");
```

Comment:
It's not supported as far as the print out of yul is concerned or in general? Because the ingot tests seem to be working so I'm a bit confused what exactly it is about ingots that `mir` doesn't support yet.

Reply:
If the input is an ingot, then the printing out of `MIR` is not supported yet. All functionality except for the pretty-printing works fine. I made the message clearer.